nsot/api/views.py

```diff
@@ -504,23 +519,41 @@ def next_network(self, request, pk=None, site_pk=None, *args, **kwargs):
         prefix_length = params.get('prefix_length')
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
-        networks = network.get_next_network(
-            prefix_length, num, strict, as_objects=False
-        )
-
-        return self.success(networks)
+        if request.method == 'POST':
+            state = 'reserved' if qpbool(params.get('reserve', False)) else 'allocated'
+            is_ip = True if prefix_length == 32 or prefix_length == 128 else False
+            networks = network.get_next_network(
+                prefix_length, num, strict, as_objects=True
+            )
+            self.allocate_networks(networks, prefix_length, site_pk, network, state=state, is_ip=is_ip)
+        else:
+            networks = network.get_next_network(
+                prefix_length, num, strict, as_objects=False
+            )
+            return self.success(networks)
+        return self.success([unicode(x) for x in networks])
 
-    @detail_route(methods=['get'])
+    @detail_route(methods=['get', 'post'])
     def next_address(self, request, pk=None, site_pk=None, *args, **kwargs):
         """Return next available IPs from this Network."""
         network = self.get_resource_object(pk, site_pk)
 
         params = request.query_params
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
-        addresses = network.get_next_address(num, strict, as_objects=False)
-
-        return self.success(addresses)
+        if request.method == 'POST':
+            addresses = network.get_next_address(num, strict, as_objects=True)
+            state = 'reserved' if qpbool(params.get('reserve', False)) else 'allocated'
+            if addresses != []:
+                if addresses[0].version == 4:
+                    prefix_length = 32
+                else:
+                    prefix_length = 128
+                self.allocate_networks(addresses, prefix_length, site_pk, network, state=state, is_ip=True)
+        else:
+            addresses = network.get_next_address(num, strict, as_objects=False)
```

Comment:
Same thing as with `next_network`, it looks like this `else` can be removed if you move the setting of `addresses` to before the if with `as_objects=True`.